### PR TITLE
Run TypeScript typechecking and eslint on Lint CI

### PR DESCRIPTION
**Pull request by somebody1234**
### Pull Request Description
- Runs `npm run typecheck` on Lint CI
- Runs `npx eslint .` on Lint CI

### Important Notes
None

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [x] The documentation has been updated, if necessary.
- [x] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.
- [x] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- All code has been tested:
  - [x] Unit tests have been written where possible.
  - [x] If GUI codebase was changed, the GUI was tested when built using `./run ide build`.

### CODE REVIEW

**Comment by somebody1234** [comment]
idk tbh. the lint seems to [fail with linker errors](https://github.com/enso-org/enso/actions/runs/4916999193/jobs/8781496456?pr=6602) around half the time. might only be an issue on certain runners?

it *might* have something to do with this PR adding a dependency...???

**Comment by somebody1234** [comment]
force-pushed because the other commits have been merged into `develop`

**Comment by somebody1234** [comment]
linker error is still happening: https://github.com/enso-org/enso/actions/runs/4922094467/jobs/8792728885?pr=6603

@mwu-tow would you happen to know what might cause this?

<blockquote><div> GitHub</div><div><strong><a href="https://github.com/enso-org/enso">Run TypeScript typechecking and eslint on Lint CI · enso-org/enso@ff20582</a></strong></div><div>Hybrid visual and textual functional programming. Contribute to enso-org/enso development by creating an account on GitHub.</div></blockquote>

File: build/cli/src/lib.rs [hunk 1]
```diff
@@ -834,7 +836,18 @@ pub async fn main_internal(config: Option<enso_build::config::Config>) -> Result
                 .run_ok()
                 .await?;
 
+            // Copied from `build` in `lib/rust/ensogl/pack/src/lib.rs`.
```
Comment (wdanilo) [inline build/cli/src/lib.rs]: 1. What is copied from `lib/rust/ensogl/pack/src/lib.rs`? The next line? Several next lines? I don't understand this comment. Anyway, if several next lines are the same as logic in `lib/rust/ensogl/pack/src/lib.rs` then this Is bad, as replicating the same logic in multiple places makes the code very hard to maintain.
2. There is no explanation what this code does. It looks complex.

**Comment by mwu-tow** [comment]
> @mwu-tow would you happen to know what might cause this?

These are random runner corruptions that happen sometimes on incremental compilation. I'll clean the runners and rerun the job.

**Comment by mwu-tow** [comment]
> @mwu-tow would you happen to know what might cause this?

These are random runner corruptions that happen sometimes on incremental compilation. I'll clean the runners and rerun the job.

**Comment by mwu-tow** [comment]
> @mwu-tow would you happen to know what might cause this?

These are random runner corruptions that happen sometimes on incremental compilation. I'll clean the runners and rerun the job.

**Comment by mwu-tow** [comment]
> @mwu-tow would you happen to know what might cause this?

These are random runner corruptions that happen sometimes on incremental compilation. I'll clean the runners and rerun the job.

File: build/cli/src/lib.rs (hunk 1 above)
Comment (mwu-tow) [reply build/cli/src/lib.rs]: The "copied" part refers to 3 lines and is not an exact copy, AFAIK. Certainly comment should be clarified.

If anything, I'd consider moving them to a separate function in `ensogl-pack`.

File: build/ci_utils/src/fs.rs [hunk 2]
```diff
@@ -138,6 +138,12 @@ pub fn remove_symlink_dir_if_exists(path: impl AsRef<Path>) -> Result {
     }
 }
 
+/// Create a symlink, or remove and recreate it if it already exists.
+pub fn create_or_update_symlink(src: impl AsRef<Path>, dst: impl AsRef<Path>) -> Result {
```
Comment (mwu-tow) [inline build/ci_utils/src/fs.rs]: Please rename it to `create_or_update_symlink_dir`. The directory symlinks and regular file symlinks are handled differently on Windows.

File: app/ide-desktop/lib/types/modules.d.ts [hunk 3]
```diff
@@ -2,6 +2,19 @@
...
  * This file MUST NOT `export {}` for the modules to be visible to other files. */
 
+// Required because this is a build artifact, which does not exist on a clean repository.
+declare module '*/build.json' {
+    interface BuildInfo {
+        commit: string
+        version: string
+        engineVersion: string
+        name: string
+    }
```
Comment (mwu-tow) [inline app/ide-desktop/lib/types/modules.d.ts]: Isn't this duplicating the definition from [app/ide-desktop/lib/types/globals.d.ts](https://github.com/enso-org/enso/blob/9c15ba06bf22bed36fc69110c5b375699858d2c7/app/ide-desktop/lib/types/globals.d.ts#L22)? Can we somehow avoid this?

Comment (somebody1234) [reply app/ide-desktop/lib/types/modules.d.ts]: good catch, i think the interfaces in `globals.d.ts` are globally accessible so i should be able to just remove the definition

Comment (somebody1234) [reply app/ide-desktop/lib/types/modules.d.ts]: maybe not... should be able to import it though

**Comment by somebody1234** [comment]
reviews: addressed
fixes: pushed
`./run git-clean; ./run lint` still works locally: yes

**Review by mwu-tow — APPROVED** [review]
Great! 👍

**Review by wdanilo — APPROVED** [review]

**Comment by PabloBuchu** [comment]
QA 🟢 I just see 1 warning in console though
```
TRACE ide_ci::program::command: npx⚠️ <START>
 INFO ide_ci::program::command: npxℹ️
 INFO ide_ci::program::command: npxℹ️ /Users/pawelbuchowski/projects/enso/app/ide-desktop/lib/dashboard/src/authentication/src/fileInfo.ts
 INFO ide_ci::program::command: npxℹ️   11:26  warning  '_extension' is defined but never used  @typescript-eslint/no-unused-vars
 INFO ide_ci::program::command: npxℹ️
 INFO ide_ci::program::command: npxℹ️ ✖ 1 problem (0 errors, 1 warning)
 INFO ide_ci::program::command: npxℹ️
TRACE ide_ci::program::command: npxℹ️ <ENDUT>
TRACE ide_ci::program::command: npx⚠️ <ENDUT>
```

**Comment by sylwiabr** [comment]
@somebody1234 what is the status? do we want to merge it now? Are we sure it will not destabilize CI?

**Comment by somebody1234** [comment]
merging with develop to minimize the chance of issues

**Comment by somebody1234** [comment]
@sylwiabr not sure if the "clean build" flag worked - but the lint check seems to be consistently passing now. that doesn't guarantee CI won't fail again, but i'm not sure what else can be done to make sure this 100% works for sure

**Comment by somebody1234** [comment]
@PabloBuchu @indiv0 do either of you want to review this? the changes to the TS code should be fairly minor

**Comment by PabloBuchu** [comment]
QA 🟢 look fine. I'll merge it now

**Review by PabloBuchu — APPROVED** [review]

**Pull request merged**